---
title: 'Chapter 5: Genetic Spatial Structure in Murray Cod'
author: "Alan Couch"
date: "`r format(Sys.time(), '%a %b %d %X %Y')`"
output: word_document
---

```{r, "Project_Template_and_Knitr", echo=FALSE, warning=FALSE}
#The following 2 and 4 lines are needed if knitr is to work with ProjectTemplate.
require(knitr)
if (basename(getwd()) == "src") setwd("..") #needed to get knitr to work with project template
library('ProjectTemplate') #All projectTemplates need this up front
load.project() #All projectTemplates need this up front
```

```{r "Set_Global_Options", echo=FALSE, warning=FALSE}
options(width=200)
opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)

knitr::opts_chunk$set(fig.width=15, fig.height=15,fig.path="../graphs/") #Make Figures larger and save them
```
This document includes methods, results and possibly some discussion dot points for the genomics and larval dispersal chapter.  

```{r "LoadLibraries"}
source("http://addictedtor.free.fr/packages/A2R/lastVersion/R/code.R")# load code of A2R function
library(ggplot2)
library(ggdendro)
library(ape)
library(dendextend)
library(Hmisc)
library(ade4)
```



##Method
 The larvae used in the dispersal analysis were collected in 2011,2012,2013 from 6 sites.
 
 Age was determined: 
 
 Species seperation
 
 Clades and r apps.
 
 DaRT as described...
 
 PCR would bias representation. p67 LFC
 
 Wilmott Skill Metric p.60 LFC - any relevence?
 
```{r}
row.names(larv)<-larv$Label
```

##Results

###Polymorphisms Examined

Dart sequencing on Next Generation Sequencing (NGS) platforms was used to reduce genome complexity allowing an intelligent selection of genome fraction corresponding predominantly to active genes. This selection was achieved through the use of a combination of Restriction Enzymes which separate low copy sequences (most informative for marker discovery and typing) from the repetitive fraction of the genome. The advantage of the DArTseq technique is the very high marker densities that were obtained which allows high resolution mapping and detailed genetic dissection of traits. 

```{r }
uniqueSNPS<-unique(DMac14.1567snps$X..1)
lengthDNA<-median(as.numeric(nchar(as.character(DMac14.1567snps$X..2))))
```
Of the 21076 alleles, the number of unique *Maccullochella* snps analysed in the DaRT sequences was `r length(uniqueSNPS)-2`. The base pairs that had mutated we as follows:

```{r table2, echo=FALSE, message=FALSE, warnings=FALSE, results='asis'}
tabl <-"
|Was |   A |  C  |  G  |  T  | Totals|
|----|-----|-----|-----|-----|-------|
|A   |     |424  |1629 |422  |2475   |
|C   |491  |     |447  |1893 |2831   |
|G   |1790 |432  |     |541  |2763   |
|T   |457  |1570 |442  |     |2469   |
|----|-----|-----|-----|-----|-------|
|Tot |2738 |2426 |2518 |2856 | 10538 |
"
cat(tabl) # output the table in a format good for HTML/PDF/docx conversion
```



```{r eval=FALSE, echo=FALSE}
#regex etc to pulls a,c,t,g for counts.
dfBPchange<-DMac14.1567snps$X..3
dfBPchange<-dfBPchange[-c(1:5)]
#head(dfBPchange)
dfBPchange<-as.data.frame(dfBPchange)
dfBPchange<-gsub(".*:","",dfBPchange$dfBPchange) #remove all position numbers before the ':'
#dfBPchange<-gsub(" ","",dfBPchange$dfBPchange) #remove all spaces
dfBPchange<-as.data.frame(dfBPchange)
dfBPchange<-dfBPchange[!(dfBPchange$dfBPchange==""), ]
dfBPchange<-as.data.frame(dfBPchange)
table(dfBPchange)
# Preferably display as a cross tab.
```

The polymorphisms examined included SNPs and small indels in restriction enzyme recognition sites, and while none were observed,  larger insertions/deletions in restriction fragments could also have been utilised. The polymorphisms selected for genotyping were all snps. 

The length of the fragments of DNA in this sequencing was `r lengthDNA` base pairs. The position of the polymorphism along the `r lengthDNA` base pair DNA fragment can be seen in the following histogram

```{r}
polymorphismPositionHistogram<-hist(as.numeric(DMac14.1567snps$X..4),main= "Histogram of SNP Position in DaRT seq 69 Base Pair Fragment", xlab="SNP Position on Fragment - BIN", breaks = 13, col = "lightblue", border = "pink",cex.main = 2.5,cex.lab=2) #a plot of the relative position of the polymorphism along the 69 base pair fragment of DNA.
#plot(polymorphismPositionHistogram)
```


###All *Maccullochella* Larvae

A dendrogram of all the larvae allows examination of the relationships between all the larvae. In the first instance this is neccessary to ensure there are no species other than Murray cod for the subsequent analysis.

```{r "All_Maccullochella_Larvae", echo=TRUE}

MacDm <- dist(allsnps) #Create distance matrix all Maccullochella larvae
#heatmap(as.matrix(MacDm)) #Make a heat map
MacHC <- hclust(MacDm) #make a heirarchical cluster
plot(color_labels(MacHC, k = 5), main="All Maccullochella Larvae Clades",cex.main = 2.5,cex.lab=0.75) 
#Plot the cluster dendrogram
```

As it turns out there are multiple *Maccullochella* species. All the larvae collected fall into one of four distinct clades. The above dendrograms shows all larvae (mostly Murray cod) and two known Trout cod controls, and what appears to be F1 and F2 hybrids between the two species. There were 4 hybrid and no pure trout cod larvae detected in the 92 larvae in 2013 caught and sequenced from the river. This is an interesting finding and detail can be found in the Trout Cod Appendix if the reader is interested.

So we now use the set with the non-Murray cod removed to identify Murray cod clades and conduct the remainder of the analysis on the Upper Murrumbidgee Murray cod larvae only.  The Trout cod and the hybrids are easily identified and eliminated from the data to ensure we are looking only at the Murray cod larvae.

###Murray Cod Larvae - 2013

First we look at a heat map and dendrogram of Murray cod larvae having excluded Trout cod and hybrids data from the data frame. 
```{r, "Murray_Cod_Larvae_Only", echo=TRUE}
MCdm <- dist(MCsnps)#Create a Murray Cod Only distance matrix
heatmap(as.matrix(MCdm), main="2013 Murray Cod Larvae Genotype Heat Map" ,cex.main = 2.5,cex.lab=2)#Heat map
MChc <- hclust(MCdm) #Cluster
#Plot it
plot(MChc,main=expression(paste("Genetic Relationships - ", italic("Maccullochella peelii"), " larvae 2013 cohort ")),cex.main = 2.5,cex.lab=2.0)
```

So there is some structure evident among the Murray cod genotypes. There is not a large difference in the dissimilarity but there is some. In most cases very closely related larvae have been collected at the same spot over the same period of a few of days. Pairs or clusters of larvae with a height of less than 26.5 are likely siblings. But in some cases very closely related pairs of larvae have turned up at different sites. Sometimes over time frames that seem unlikely or suggest the larvae have travelled upstream. However, must remember that it is siblings - not the same fish - that is caught so this might just represent the 'smear' of larvae along the river after dispersing from the nest.

The distance matrix suggests a very low genetic diversity in the population of Murray cod sampled with most of the distances around 0.03. Given this, can we be sure that the most closely related larvae are siblings? 

```{r, "A2R_Dendrogram"}

#Colour Dendrogram
#A2Rplot(MChc, k = 12, boxes = FALSE, col.up = "gray50")
dend1 <- color_branches(MChc, k = 12)
dend1 <- color_labels(dend1, k = 12)#Use h or k
plot(dend1, main=expression(paste("Genetic Relationships - ", italic("Maccullochella peelii"), " larvae 2013 cohort ")),cex.main = 2.5,cex.lab=2)
```

While the Murray cod larvae are all very closely related, they can now be resolved into 12 clades.These clades represent 12 'extended family groups' and the 3 clades represent the three higher level (race) clades in the river.

```{r, "ExtractClades", echo=TRUE}
cladeNo<-cutree(dend1,k=12) #This is like using:dendextend:::cutree.dendrogram(dend1,h=70) h or k can be specified
cladeNo<-as.data.frame(cladeNo)
#For example:
#cladeNo[c(20:30),] #The clades are numbered by default. So I need to name them something sensible for subsequent analysis.
```

Extract three higher level clades
```{r "extractRaceClades", echo=TRUE}
dend2 <- color_labels(MChc, k = 3)#Use h or k
raceCladeNumber<-cutree(dend2,k=3) #This is like using:dendextend:::cutree.dendrogram(dend1,h=70) h or k can be specified
raceCladeNumber<-as.data.frame(raceCladeNumber)
#raceCladeNumber[,] #The clades are numbered by default. So I need to name them something sensible for subsequent analysis.

larvalClades<-merge(cladeNo,raceCladeNumber, by="row.names")
row.names(larvalClades)<-larvalClades$Row.names
larvalClades$Row.names<-NULL

#So the list of larval clades no and raceCladeNumber -> database.

#write.csv(DF named something,file= ".\\forDataBase\\geneticGroupings.csv") #This line is hashed out because it can not be done from within knitr and only needs to be done once (correctly) as the data winds up in the database for all chapters to use.

```

### Interannual Differences
Larvae were collected from three years.

#### Genetics of 2013 Cohort

#### Genetics of 2012 Cohort

#### Genetics of 2011 Cohort

### Parentage Analysis
Colony etc. Page 68 LFC

half sibs: sneaker males to nests?

Confirmation of siblings determined by DaRT data?

Any sibs across years? Siblings to be considered withing year and possibly, but less likely across years.

Siblingness supported by the temporal and spatial proximity alone (next chapter) which can predict genotype siblingness with confusion error of....

###AMOVA
- Bernd's data munging etc
-to support earlier assertion that they are 'very closely related' from dart data alone.

### PLD Kernel (LFC p.64)

##Discussion (points only)
* Evidence for local recruitment. Re recruitment to local area.

* IBD quite marked compared to some marine species. See salmon over thousands of kilometres (same r squared) [ref: powerpoint slide] p.65 LFC

* Eel tramps p110 LFC

*Natal Signatures p80 and p31 LFC

p68 and Gst and F st p72 LFC


##Code Chunks in this Document

```{r "Include_Chunk_Labels_and_Session Information"}
all_labels()

#Session Information
sessionInfo()

```