
---
title: "Upper Murrumbidgee Larval Cod Dispersal Analysis"
author: "Alan Couch"
date: "`r format(Sys.time(), '%a %b %d %X %Y')`"
output: word_document
---



```{r, "Project_Template_and_Knitr", echo=TRUE, warning=FALSE}
#The following 2 and 4 lines are needed if knitr is to work with ProjectTemplate.
require(knitr)
if (basename(getwd()) == "src") setwd("..") #needed to get knitr to work with project template
library('ProjectTemplate') #All projectTemplates need this up front
load.project() #All projectTemplates need this up front
```

```{r "Set_Global_Options", echo=FALSE, warning=FALSE}
options(width=200)
opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)

knitr::opts_chunk$set(fig.width=15, fig.height=15,fig.path="../graphs/") #Make Figures larger and save them
```

```{r "LoadLibraries", echo=TRUE}
source("http://addictedtor.free.fr/packages/A2R/lastVersion/R/code.R")# load code of A2R function
library(ggplot2)
library(ggdendro)
library(ape)
library(dendextend)
library(Hmisc)
library(ade4)
```
This document maps out the analysis of the dispersal of Murray cod larvae in the Murriumbidgee river. It includes sampling, larval morphological characteristics, genome profile and a natural biogeochemical markers. The larvae used in the dispersal analysis were collected in 2013 from 6 sites. The data used for ageing were collected over the three years 2011-2013.

![Map of Study Area](../images/M2GStudyArea.jpg)


Calculate Some Additional Parameters
------------------------------------
This section is to add some calculated variables to the data. In particular to add:
 
  Age from Otolith Length: 74.308*[MeanOtolithLength]-4.44361
  
  Hatch DoY :  [Day of Year Caught]-[Age From Otolith Length]
  
  Incubation: 20.67-0.667*[WaterTemp(DegC) Mean]
  
  Spawning:[Hatch]-[Incubation]

These additional parameters will be used to estimate distances that the larvae have dispersed based on the number of days available to them since leaving the nest and the day and location of collection. In turn The time available will be used to model the most likely distance travelled by the larvae.

```{r "Additional_Calculations", echo="TRUE"}

larv$ageOL<-74.308*larv$Mean.Otolith.Length.is.in.Millimetres.for.comparison.with.Adults-4.44361
larv$hatchdoy<-larv$Day.of.Year-larv$ageOL
larv$incTime<-20.67-0.667*larv$WaterTemp.DegC..Mean
larv$spawn<-larv$hatchdoy-larv$incTime

larv[c(20:30),c(119:122)] #just to see all OK
```

But before that we need to establish the genetic identies so that we can test the assumption that genetic distance is positively corelated with geographic distance between individuals.


All Maccullochella Larvae
-------------------------
If we look at all the larvae collected we see four distinct clades.The Trout cod and the hybrids are easily identified and eliminated from the data to ensure we are looking only at the Murray cod larvae.

```{r "All_Maccullochella_Larvae", echo=TRUE}

MacDm <- dist(allsnps) #Create a distance matrix for all Maccullochella larvae

heatmap(as.matrix(MacDm)) #Make a heat map

MacHC <- hclust(MacDm) #make a heirarchical cluster

plot(color_labels(MacHC, k = 5), main="All Maccullochella Larvae") #Plot the cluster dendrogram
```

The above dendrograms shows Murray cod and two known Trout cod controls and what appears to be F1 and F2 hybrids. There were only 4 hybrid and 0 pure trout cod larvae deteced in the 92 larvae caught and sequenced. 

It will be interesting to mito-sequence the trout cod and determine the species of the male and female parent. I expect that the female is the Trout cod given the scarcity of trout cod compared with Murray cod and the mate pressure that must exist. 

So we now remove the non- Murray cod to identify Murray cod haplogroups and conduct the remainder of the analysis on the Upper Murrumbidgee Murray cod larvae only.  

Murray Cod Larvae
-----------------
```{r, "Murray_Cod_Larvae_Only", echo=TRUE}
MCdm <- dist(MCsnps)#Create a Murray Cod Only distance matrix

heatmap(as.matrix(MCdm))#Heat map

MChc <- hclust(MCdm) #Clutser
#Plot it
plot(MChc,main="Maccullochella peelii Larvae")
```

In most cases very closely related larvae have been collected at the same spot over the same couple of days. But in some cases very closely related pairs of larvae have turned up at different sites. Sometimes over time frames that seem unlikely or suggest the larvae have travelled upstream. However, must remember that it is siblings - not the same fish - that is caught so this might just represent the 'smear' of larvae along the river after dispersing from the nest.

The distance matrix suggest a very low genetic diversity in the population of Murray cod sampled. Given this, how to determine if the closely related larvae are siblings? 

```{r, "A2R_Dendrogram"}

#Colour Dendrogram
#A2Rplot(MChc, k = 12, boxes = FALSE, col.up = "gray50")
dend1 <- color_labels(MChc, k = 12)#Use h or k
plot(dend1)
```

```{r, "Another_Dendrogram", echo=TRUE}
# basic option
ggdendrogram(MChc, rotate = TRUE, size = 4, theme_dendro = FALSE)

```
The Murray cod can be resolved into 12 clades. Can I call these Haplogroups? Is 12 the right sort of cutoff? Should there be more?
```{r, "ExtractClades", echo=TRUE}
#dend1 <- color_branches(MChc, k = 12)
# Get the package:
#cutree(dend1,h=70) # it now works on a dendrogram. Use h or k
# It is like using:
#dendextend:::cutree.dendrogram(dend1,h=70)

cladeNo<-cutree(dend1,k=12)
cladeNo<-as.data.frame(cladeNo)
#For example:
cladeNo[c(20:30),] #The clades are numbered by default.
#write.csv(format(cladeNo, digits=2, trim=TRUE), file="table1.csv", row.names=TRUE, quote=FALSE)
```

Trout Cod Larvae
----------------
```{r, "Trout_Cod_Dendrograms", echo=TRUE}
#and a Trout Cod 
TCdm <- dist(TCsnps)
#Heat map
dataMatrix <- as.matrix(TCdm)
heatmap(dataMatrix)
#cluster
TChc <- hclust(TCdm)
#Plot it
#plot(TChc, main="Maccullochella macquariensis larvae")
A2Rplot(TChc, k =3, boxes = FALSE, col.up = "gray50",main = "Maccullochella macquariensis larvae")

```


Upper Murrumbidgee Larval Cod Genetic and Geographic Distances
==============================================================
This test corelates the physical distances of the nests of the larvae with genetic distance of larvae from the Murrumbidgee collected in 2013 from 6 sites.
By iterating the mantel test using distance matrices generated over varying nest distances and finding the most highly corelated provides a best estimate of the distance a nest site is.

# read chunk (does not run code)
```{r echo=FALSE}
read_chunk('IteratedMantel.R')
```

# run the variablesXY chunk and use the variables it creates
```{r first}
<<IM>>
```
In this case it is the distance above the collection site if the larvae 'drift' `r BestNestEst` metres per day between leaving the nest and being collected at sampling site. 

There is a possibiltiy that if this was done with mitochondrial sequences and downstream distance matrix the best estimate for the female upward migration might be found if iterate say: 1000-20000m. A similar approach using the chemotype might allow use to do something similar with it and geo distance.

```{r "Distance_Matrices_and_Ordering", echo=TRUE}
# To create a distance
larv$nestdist<-larv$Distance.to.Angle.Crossing..m.-(BestNestEst*(larv$Day.of.Year-(larv$hatchdoy+7)))

#remove larvae that do not have genetic analysis done.
#Creat a MCsnps set with row names as a column.
MCchecklist<-row.names(MCsnps)
MCchecklist<-as.data.frame(MCchecklist)# 93 records

#remove a few more anomolies
MCchecklist1 <- as.data.frame(MCchecklist[-c(1:7), ])
# Keep every record in larv that is also in MCchecklist (i.e., the intersection).

larv_intersection <- larv[larv$Label %in% MCchecklist$MCchecklist,]
#Thanks: https://heuristically.wordpress.com/2009/10/08/delete-rows-from-r-data-frame/

larv<-larv_intersection
larv_intersection<-NULL

###########
# Recreate Genetic Distance
# Create a Murray Cod distance matrix
MCdm<-MCsnps[-c(1:7),]
MCdm <- dist(MCdm)
MCdm<-as.matrix(MCdm)
MCdm<-as.data.frame(MCdm)
#This is to be used for plotting
###########

geodist<-data.frame(larv$Label,larv$nestdist)
row.names(geodist)<-geodist[,1]
geodist$larv.Label<-NULL
geodist<-na.omit(geodist)
geodist1000<-geodist #save this estimate for haplogroups distance plot (after the Iterated Mantel has changed it)

GeoDistMat<-dist(geodist)
GeoDistMathm <- as.matrix(GeoDistMat)
heatmap(GeoDistMathm)
geoclust<-hclust(GeoDistMat)
plot(geoclust)

#make sure both matrices are in correct order - rows and cols
#First sort MCdm

MCdm<-as.data.frame(MCdm)
MCdm$sort<-row.names(MCdm)
MCdm <- MCdm[order(MCdm$sort),]#sort row order
MCdm$sort<-NULL
MCdm<-MCdm[,order(names(MCdm))]#sort column order
MCdm<-as.matrix(MCdm)

#Second sort GeoDist
GeoDistMathm<-as.data.frame(GeoDistMathm)
GeoDistMathm$sort<-row.names(GeoDistMathm)
GeoDistMathm <- GeoDistMathm[order(GeoDistMathm$sort),]#sort row order
GeoDistMathm$sort<-NULL
GeoDistMathm<-GeoDistMathm[,order(names(GeoDistMathm))]#sort column order
GeoDistMathm<-as.matrix(GeoDistMathm)

larv1<-larv#save this estimate for haplogroups distance plot (after the Iterated Mantel has changed it)
```
Now that these various matrices, class 'dist' objects are created we can proceed for plot.

Plot and Correlate genetic and geographic distance matrices
-----------------------------------------------------------

```{r "Plots_and_Correlation", echo=TRUE}
#Linear Regression Model
reg=lm(MCdm[lower.tri(MCdm)]~GeoDistMathm[lower.tri(GeoDistMathm)])
summary(reg)
plot(GeoDistMathm[lower.tri(GeoDistMathm)],MCdm[lower.tri(MCdm)])
abline(reg)
# Correlations with significance levels
rcorr(GeoDistMathm[lower.tri(GeoDistMathm)],MCdm[lower.tri(MCdm)])#(x, type="pearson") # type can be pearson or spearman
```

So there is some small but significant correlation between genetic distance and geographic distance in the Murray cod sampled. 

Mantel Test
-----------
```{r "Mantel_Test", echo=TRUE}
mant<-mantel.rtest(as.dist(GeoDistMathm), as.dist(MCdm), nrepet = 9999)
mant
#Check all is in order
as.matrix(GeoDistMathm)[1:5, 1:5]
as.matrix(MCdm)[1:5, 1:5]
```
Based on these results, we can reject the null hypothesis that these two matrices, spatial distance and genetic distance, are unrelated with alpha = `r mant$pvalue`.  The observed correlation, `r mant$obs`, suggests that the matrix entries are positively associated.  So smaller differences in genotype are generally seen among pairs of larvae that are from nests estimated to be geographicallyclose to each other, rather than nests which are estimated to be further away from each other.   Note that since this test is based on random permutations, the same code will always arrive at the same observed correlation but rarely the same p-value. 

The question needs to be asked, and answered, as to how the 1000m distance used in the nest distance  estimation equation was derived. This was estimated by iterating the Mantel test for each distance matrix calculated after calculating that matrix using a 'larval dispersal' distance from 1 m to 5000 metres.The highest corellation suggests that it is the distance that best represents the average distance that larvae disperse. The curve produced from the estimations is as follows.



Look at Clades Over the River Reach
-----------------------------------
```{r "Clades_and_Location", echo =TRUE}


larv1<-merge(larv,cladeNo, by="row.names")
rownames(CladeNamesToMerge) <- CladeNamesToMerge[,1]
CladeNamesToMerge$Label<-rownames(CladeNamesToMerge)
larv2<-merge(larv1,CladeNamesToMerge, by="Label")
plot(larv2$nestdist,larv2$clade, xlab="Nest Distance (m)", ylab = "Clade")
#merged<-merge(Haplogroups,geodist1000, by="row.names")      
#plot(merged$larv.nest,merged$Haplogroups)
plot2 <- ggplot(larv2, aes(nestdist,clade))
plot2 + geom_point(alpha = 2/3) +geom_boxplot()+labs(title = "Clades Nests and their Position on River")+ labs(x="Nest Location") +labs(y = "Clade")+ theme(axis.text.x=element_text(angle=90))
```
This suggests that all clades except one (hT1) exist below a barrier around 10000m but that only three clades (hM1,hL1 and hK2)  exist above and below this barrier. On possible explanation that might be inferred from this is that adults migrating upstream for spawning are prevented from doing so by a barrier at 10000m mark but that larvae produced above the barrier are able to disperse and so are represented downstream. It is noteable that the barrier and the big gap in apparent nests from 6000m to 14000m corresponds with the Tharwa sand slug - a long stretch of sand that has previously been the subject of remedial work because it has been believed to be a barrier to cod migration (ref).  

![Nest Along the River](../images/Dashboard1.jpg)


Natural BioGeochemistry Markers
===============================
The first effort is based on the location of collection. Later, if there is a corelation between 'chemotype distance' and geographic distance we will look at the possibility of using an offset downstream of the origin of the mother. It is not likely possible as with larvae we could use larval 'drift' age days to get infrormation about each larvae. We can do this with the mothers but can we do something with the clade groups?

Identify Important Chemistry Variables for Prediction of Site
-------------------------------------------------------------
In order to identify important variables in a multivariate dataset one can utilize machine learning methods. There are many different machine learning algorithms for different tasks. One common task is to decide if a feature vector belongs to a certain class. This can be done with a random forest [2] classifier. In order to do so, one has to train the classifier with training data first. Then the classifier can be used to predict the class of other feature vectors.http://proven-inconclusive.com/blog/machine_learning_methods_to_identify_important_variables.html.  There is no need for other tests, such as cross-validation, to get an unbiased estimate of the test set error as each tree is created with a different bootstrap sample [2].

The classifier saves information on feature importance ("importance=TRUE"). We can use this information in order to identify potentially import variables in the data set.

```{r echo=FALSE}
read_chunk('RandForestToIDImportantVariables.R')
```
# run the IM chunk (all) RandomForestChem
```{r IRandomForestGo, echo=TRUE}
<<RandomForestChem>>
```
This is now rubbish: Given that no part of the otolith can provide an out-of-bag classification rate lower that 22% it suggests that otolith microchemistry alone, over this small spatial scale, is insufficiently variable to make a good predictor of site of origin for the Larvae. It is worth noting that the otolith core provided the most accurate classification at 22.something  %. Given the accuracy of the core chemistry is sub-optimal because of the inability of the operator to accuratel identify and sample the centre of the otolith in all cases with laser ablation the possibilty remains that cleaning the data in that dataset might be improved by deleting inaccurate ablates. This is possible because each otolith was assigned an accuracy score.

A Chemotype Dendrogram and a Mantel Test.
-----------------------------------------
 Now that we have identified the best variables to use from the data we can create a dendrogram and use Mantel test to look for any relationship between chemotype and geographic distances. This first creates distance matrices for chemotype and geographic distance and then uses a mantel test to see correlation. The latter chemotypes are scaled.
```{r echo=FALSE}
read_chunk('ChemotypeDistanceCorrelations.R')
```

```{r chemotypeDistanceMatricesGo, echo=TRUE}
<<chemotypeDistanceMatrices>>
```
Based on these results, we can reject the null hypothesis that these two matrices, spatial distance and chemotype distance, are unrelated with alpha = `r mant$pvalue`.  The observed correlation, `r mant$obs`, suggests that the matrix entries are positively associated.  So differences in chemotype between pairs of larvae can be used to improve prediction of geographic distance.







Anomoly In Fish Detected
========================
In the original snps there were 5 fish that... TBC
something like this:

```{r}
DArTdm2<-DArTdm
row.names(DArTdm2)<-DArTdm2[,1]
DArTdm2$X.<-NULL
heatmap(as.matrix(DArTdm2))
DArTdm2<-as.dist(DArTdm2)
plot(hclust(DArTdm2))
```
#OR
unknowns <- dist(DArTdm)

#Make a heat map
dataMatrix <- as.matrix(unknowns)
heatmap(dataMatrix)

#Do a heirarchical cluster
unknownshc <- hclust(unknowns)
#Plot the cluster dendrogram
plot(unknownshc, main="Original DArT Matrix")

But they need labels etc.

Code Chunks in this Document
----------------------------
```{r "Include_Labels_for_all_code_Chunks"}
all_labels()

#Session Information
sessionInfo()
```